ruby/lib/gilded_rose.rb

```diff
@@ -1,68 +1,52 @@
+require "items"
+require "item_type"
+require "helper_functions"
+
 class GildedRose
+  attr_reader :items
+
+  SULFURAS = "Sulfuras, Hand of Ragnaros"
+  BRIE = "Aged Brie" 
+  BACKSTAGE = "Backstage passes to a TAFKAL80ETC concert" 
 
   def initialize(items)
     @items = items
   end
 
-  def update_quality()
-    @items.each do |item|
-      if item.name != "Aged Brie" and item.name != "Backstage passes to a TAFKAL80ETC concert"
-        if item.quality > 0
-          if item.name != "Sulfuras, Hand of Ragnaros"
-            item.quality = item.quality - 1
-          end
-        end
-      else
-        if item.quality < 50
-          item.quality = item.quality + 1
-          if item.name == "Backstage passes to a TAFKAL80ETC concert"
-            if item.sell_in < 11
-              if item.quality < 50
-                item.quality = item.quality + 1
-              end
-            end
-            if item.sell_in < 6
-              if item.quality < 50
-                item.quality = item.quality + 1
-              end
-            end
-          end
-        end
-      end
-      if item.name != "Sulfuras, Hand of Ragnaros"
-        item.sell_in = item.sell_in - 1
-      end
-      if item.sell_in < 0
-        if item.name != "Aged Brie"
-          if item.name != "Backstage passes to a TAFKAL80ETC concert"
-            if item.quality > 0
-              if item.name != "Sulfuras, Hand of Ragnaros"
-                item.quality = item.quality - 1
-              end
-            end
-          else
-            item.quality = item.quality - item.quality
-          end
-        else
-          if item.quality < 50
-            item.quality = item.quality + 1
-          end
-        end
-      end
+  def update_quality
+    items.each do |item|
+      update_item(item)
+    end
+  end
+
+  def update_item(item)
+    if item.name == SULFURAS
+    elsif item.name == BRIE
```

Comment:
Note to self - to discuss.

Reply:
`if item.name == SULFURAS then return item`

Reply:
created method for sulfuras where quality is set to 80 and does not change